instrumentation/manticore/lib/opentelemetry/instrumentation/manticore/patches/response.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Manticore
+      module Patches
+        # Module to prepend to RestClient::Request for instrumentation
+        module Response
+          # Hooks into the #.call function which is executed for any verbs like:
+          #   [:get, :post, :put, :options, :header]
+          # Executes for async and sync requests.
+          # It has direct to Manticore::Response methods and instance variables.
+          # Take great care of any unintentional mutations:
+          # [@request, @response, .on_complete] - are being invoked directly from Manticore::Response
+          def call
+            wrapped_request = OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest.new(@request)
+            attributes = request_attributes(wrapped_request).merge(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+            span = tracer.start_span("HTTP #{wrapped_request.method}", attributes: attributes, kind: :client)
+            OpenTelemetry::Trace.with_span(span) do
+              OpenTelemetry.propagation.inject(wrapped_request)
+            end
+            on_complete do |response|
+              annotate_span_with_response!(span, response)
+            end
+            # .super() somehow becomes missing for async requests. Below works for sync and async requests.
+            self.method(:call).super_method.call
+          rescue ::Manticore::ManticoreException => e
+            OpenTelemetry.logger.debug("Errored during tracing: #{e}")
+            span.record_exception(e)
+          ensure
+            span.finish
+          end
+
+          private
+
+          # @param [OpenTelemetry::Trace::Tracer.start_span] span A running span that has not been completed
+          # @param [Manticore::Response] response A @response instance object after an outgoing call is attempted
+          def annotate_span_with_response!(span, response)
+            attr = response_attributes(response)
+            span.add_attributes(attr)
+            span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response.code)
+          end
+
+          # @param [OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest] wrapped_request A WrappedRequest object that exposes some useful methods.
+          # @return [Hash] returns request attributes to be added to span
+          def request_attributes(wrapped_request)
+            uri = URI.parse(wrapped_request.uri)
+            attr = {
+              'library' => 'Manticore',
```

Comment:
I don't believe this attribute is part of the specification for http clients. You should be able to determine the library being used by the `span.instrumention_library.name`. That being said if this is useful to you currently and your backend isn't able to lookup/display/digest the `span.instrumentation_library.name` properly (It's part of OTLP spec but not necessarily all export options), I'd just prefer this attribute get namespace'd to be `manticore` specific, ie, `manticore.library_name`, so in case a generic `library` gets added to the spec later, we won't be in compliance. Wdyt?

Reply:
No problem. This can be removed.